src/js/controllers/topup.js

```diff
@@ -50,29 +60,146 @@ angular.module('copayApp.controllers').controller('topUpController', function($s
     }
   };
 
-  var createInvoice = function() {
-    $scope.expirationTime = null;
-    ongoingProcess.set('creatingInvoice', true);
-    bitpayCardService.topUp(cardId, dataSrc, function(err, invoiceId) {
+  var setTotalAmount = function(amountSat, invoiceFeeSat, networkFeeSat) {
+    satToAlternative(amountSat, function(a) {
+      $scope.amount = Number(a);
+
+      satToAlternative(invoiceFeeSat, function(i) {
+        $scope.invoiceFee = Number(i);
+
+        satToAlternative(networkFeeSat, function(n) {
+          $scope.networkFee = Number(n);
+          $scope.totalAmount = $scope.amount + $scope.invoiceFee + $scope.networkFee;
+          $timeout(function() {
+            $scope.$digest();
+          });
+        });
+      });
+    });
+  };
+
+  var createInvoice = function(data, cb) {
+    bitpayCardService.topUp(cardId, data, function(err, invoiceId) {
       if (err) {
-        ongoingProcess.set('creatingInvoice', false);
-        showErrorAndBack(gettextCatalog.getString('Could not create the invoice'), err);
-        return;
+        return cb({
+          title: gettextCatalog.getString('Could not create the invoice'),
+          message: err
+        });
       }
 
       bitpayCardService.getInvoice(invoiceId, function(err, inv) {
-        ongoingProcess.set('creatingInvoice', false);
         if (err) {
-          showError(gettextCatalog.getString('Could not get the invoice'), err);
+          return cb({
+            title: gettextCatalog.getString('Could not get the invoice'),
+            message: err
+          });
+        }
+        return cb(null, inv);
+      });
+    });
+  };
+
+  var createTx = function(wallet, invoice, message, cb) {
+    var payProUrl = (invoice && invoice.paymentUrls) ? invoice.paymentUrls.BIP73 : null;
+
+    if (!payProUrl) {
+      return cb({
+        title: gettextCatalog.getString('Error in Payment Protocol'),
+        message: gettextCatalog.getString('Invalid URL')
+      });
+    }
+
+    var outputs = [];
+    var toAddress = invoice.bitcoinAddress;
+    var amountSat = parseInt(invoice.btcDue * 100000000); // BTC to Satoshi
+
+    outputs.push({
+      'toAddress': toAddress,
+      'amount': amountSat,
+      'message': message
+    });
+
+    var txp = {
+      toAddress: toAddress,
+      amount: amountSat,
+      outputs: outputs,
+      message: message,
+      payProUrl: payProUrl,
+      excludeUnconfirmedUtxos: configWallet.spendUnconfirmed ? false : true,
+      feeLevel: configWallet.settings.feeLevel || 'normal'
+    };
+
+    walletService.createTx(wallet, txp, function(err, ctxp) {
+      if (err) {
+        return cb({
+          title: gettextCatalog.getString('Could not create transaction'),
+          message: bwcError.msg(err)
+        });
+      }
+      return cb(null, ctxp);
+    });
+  };
+
+  var parseAmount = function(wallet, cb) {
```

Comment:
Maybe this function should be `calculateAmount`...

```
if (useSendMax)  {
  ...
} else 
  return cb(null, amount);
```

and on the result of the function use `txFormatService.parseAmount`